sql/core/src/test/scala/org/apache/spark/sql/sources/v2/FileDataSourceV2Suite.scala

```diff
@@ -0,0 +1,179 @@
+        spark.read.format(dummyParquetReaderV2).load(path)
+      }
+      assert(exception.message.equals("Dummy file reader"))
+    }
+  }
+
+  test("Fall back to v1 when reading file with write only FileDataSourceV2") {
+    val df = spark.range(1, 10).toDF()
+    withTempPath { file =>
+      val path = file.getCanonicalPath
+
+      // Dummy File writer should fail as expected.
+      val exception = intercept[AnalysisException] {
+        df.write.format(dummyParquetWriterV2).save(path)
+      }
+      assert(exception.message.equals("Dummy file writer"))
+
+      df.write.format(parquetV1).save(path)
+      // Reading file should fall back to v1 and succeed.
+      checkAnswer(spark.read.format(dummyParquetWriterV2).load(path), df)
+    }
+  }
+
+  test("Fall back read path to v1 with configuration DISABLED_V2_FILE_DATA_SOURCE_READERS") {
+    val df = spark.range(1, 10).toDF()
+    withTempPath { file =>
+      val path = file.getCanonicalPath
+      df.write.format(parquetV1).save(path)
+      withSQLConf(SQLConf.DISABLED_V2_FILE_DATA_SOURCE_READERS.key -> "foo,parquet,bar") {
+        // Reading file should fall back to v1 and succeed.
+        checkAnswer(spark.read.format(dummyParquetReaderV2).load(path), df)
+      }
+
+      withSQLConf(SQLConf.DISABLED_V2_FILE_DATA_SOURCE_READERS.key -> "foo,bar") {
+        // Dummy File reader should fail as DISABLED_V2_FILE_DATA_SOURCE_READERS doesn't include it.
+        val exception = intercept[AnalysisException] {
+          spark.read.format(dummyParquetReaderV2).load(path)
+        }
+        assert(exception.message.equals("Dummy file reader"))
+      }
+    }
+  }
+
+  test("Fall back write path to v1 with configuration DISABLED_V2_FILE_DATA_SOURCE_READERS") {
+    val df = spark.range(1, 10).toDF()
+    withTempPath { file =>
+      val path = file.getCanonicalPath
+
+      withSQLConf(SQLConf.DISABLED_V2_FILE_DATA_SOURCE_WRITERS.key -> "foo,bar") {
+        // Dummy File writer should fail as expected.
+        val exception = intercept[AnalysisException] {
+          df.write.format(dummyParquetWriterV2).save(path)
+        }
+        assert(exception.message.equals("Dummy file writer"))
+      }
+
+      withSQLConf(SQLConf.DISABLED_V2_FILE_DATA_SOURCE_WRITERS.key -> "foo,parquet,bar") {
+        // Writing file should fall back to v1 and succeed.
+        df.write.format(dummyParquetWriterV2).save(path)
+      }
+
+      // Validate write result with [[ParquetFileFormat]].
+      checkAnswer(spark.read.format(parquetV1).load(path), df)
+    }
+  }
+
+  test("InsertIntoTable: Fall back to V1") {
+    val data = (100 until 105).map(i => (i, -i))
+    val data2 = (5 until 10).map(i => (i, -i))
+    withTempPath { file =>
+      val path = file.getCanonicalPath
+      withTempView("tmp", "tbl") {
+        spark.createDataFrame(data).toDF("i", "j").createOrReplaceTempView("tmp")
+        spark.createDataFrame(data2).toDF("i", "j").write.format(parquetV1).save(path)
```

Comment:
`xxx.write.parquet(...)`, then we don't need the `parquetV1 `

Reply:
Mase sense. I was trying to make sure it is using V1 format, which seems unnecessary.